### PR TITLE
[Merged by Bors] - feat: new noisy check

### PR DESCRIPTION
**Pull request by adomani**
This PR restructures a little the noisy tests, exploiting the new `lake` replay feature.

Moreover, it catches previously uncaught noise:
* see #12884 for a noisy file that passes the old CI;
* see #12885 for the same noisy file that does not pass the new CI.

Note that linter warnings and `sorry`s are also not caught by the old CI:
* #12910;
* #12912;
* #12913.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: .github/workflows/bors.yml [hunk 1]
```diff
@@ -266,9 +266,13 @@ jobs:
       - name: check for noisy stdout lines
         id: noisy
         run: |
-          grep --after-context=1 "stdout" stdout.log && ret=0
-          grep --after-context=1 "stderr" stdout.log && new=0
-          if [ "${ret}" == "0" ] || [ "${new}"  == "0" ]; then exit 1; fi
+          buildMsgs="$(lake build Mathlib Archive Counterexamples |
+            awk '!($1 ~ "^[[?0-9]*/[?0-9]*]"){ print $0 }')"
```
Comment (semorrison) [inline .github/workflows/bors.yml:270]: Could you add a comment explaining what this `awk` invocation is doing?

Perhaps also a comment above buildMsgs that explains we are taking advantage of lake's replay.

Comment (adomani) [reply .github/workflows/bors.yml:276]: Comments added: let me know if this is too much/too little!

Comment (jcommelin) [reply .github/workflows/bors.yml:276]: LGTM

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -273,9 +273,19 @@ jobs:
...
-          grep --after-context=1 "stderr" stdout.log && new=0
-          if [ "${ret}" == "0" ] || [ "${new}"  == "0" ]; then exit 1; fi
+          buildMsgs="$(
+            ##  we exploit `lake`s replay feature: since the cache is present, running
+            ##  `lake build` will reproduce all the outputs without having to recompute
+            lake build Mathlib Archive Counterexamples |
+            ##  we filter out the output lines that begin with `[xx/yy]`, where xx, yy
+            ##  are either numbers or ?
+            ##  we keep the rest, which are actual outputs of the files
+            awk '!($1 ~ "^[[?0-9]*/[?0-9]*]"){ print $0 }')"
```
Comment (semorrison) [inline .github/workflows/build.yml:283]: This is very likely to break in rc2, after Mac is finished with https://github.com/leanprover/lean4/pull/4127. We'll just adapt as needed.

**Comment by semorrison** [comment]
Let's try it out!

bors merge

File: .github/workflows/build.yml (hunk 2 above)
Comment (adomani) [reply .github/workflows/build.yml:283]: Ok, I'm happy to monitor what needs to be changed with rc2!

As is, CI is not really testing for noisy files anymore.

**Pull request closed without merging**